src/command.c

```diff
@@ -670,6 +670,14 @@ static bool cmd_shutdown_bmda(target_s *t, int argc, const char **argv)
 }
 #endif
 
+/*
+ heapinfo can be used to quickly test a system with different heap and stack values, to see how much heap and stack is needed.
+ - user sets up values for heap and stack using "mon heapinfo"
+ - when the target boots, crt0.S does a heapinfo semihosting call to get these values for heap and stack.
+   See newlib/libc/sys/arm/crt0.S "Issue Angel SWI to read stack info"
+ - if the system crashes increase heap or stack
+ */
+
```

Comment:
Please drop the extra new line so InteliSense and other tools see the comment as documentation for the function.

Reply:
OK.
Every command has a one-line help message stored in cmd->help.
Maybe some bytes could be saved if a command, instead of printing a help message and exiting, could signal back: print the one-line help message for my command and exit.

Reply:
That could work for a simple version of the implementation - though the parens around the argc check need dropping and you'd need to use something other than `bool` for `retval` as `bool` is not allowed to have an inspectable numerical value other than 0 and not-0 w/ not-0 being coerced to 1 when converted back to integer.

Reply:
I like the idea, but this will require changing the `bool (*cmd_handler_fn)(target_s *target, int argc, const char **argv)` function-signature and consequently all the commands just so they can return 2. I'm more inclined towards using identical message strings. See PR1522 for an example of gcc/ld.bfd folding many identical constant string literals.